addons/website_livechat/models/discuss_channel.py

```diff
@@ -37,7 +37,7 @@ def _to_store(self, store: Store):
             try:
                 country_id = visitor.partner_id.country_id or visitor.country_id
                 channel_info['visitor'] = {
-                    'name': visitor.partner_id.name or visitor.partner_id.display_name or visitor.display_name,
+                    'name': visitor.partner_id.name or visitor.partner_id.display_name or visitor.display_name or f"Visitor #{visitor.id}",
```

Comment:
Don't forget to translate `Visitor #{id}`, by wrapping into `_()` and using `%s`